### PR TITLE
NumberInput

### PR DESCRIPTION
**Pull request by Oli8**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -100,6 +100,7 @@ Any donation is greatly appreciated :heart:
 - Form
 - Checkbox
 - Input
+- NuberInput
```
Comment (johanesa) [inline README.md:103]: Order alphabetically

File: src/lib/components/Form/NumberInput/NumberInput.svelte [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        disabled={!canDecrement}
+      />
+    {/if}
+    <label>
+      {label}
+      <input {...attr}
+             inputmode="numeric"
+             bind:value={formattedValue}
+             class:border-danger={!isValid && value}
+             class:border-success={isValid && value && valid}
```
Comment (johanesa) [inline src/lib/components/Form/NumberInput/NumberInput.svelte:16]: what is `isValid` and `valid`

File: src/lib/components/Form/NumberInput/NumberInput.svelte [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        clickHandler={() => value -= step}
+        type={controlsType}
+        disabled={!canDecrement}
+      />
+    {/if}
+    <input {...attr}
+           inputmode="numeric"
+           bind:value={formattedValue}
+           class:border-danger={!isValid && value}
+           class:border-success={isValid && value && valid}
```
Comment (johanesa) [inline src/lib/components/Form/NumberInput/NumberInput.svelte:46]: same

File: src/lib/components/Form/NumberInput/NumberInput.svelte [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+$: if (valid) {
+  isValid = valid(value);
+}
+
+let attr: Attributes;
+$: {
+  attr = getDomAttributes({
+    props: $$props,
+    classes,
+    toOmit: ['block', 'value', 'valid', 'noResize', 'min', 'max']
```
Comment (johanesa) [inline src/lib/components/Form/NumberInput/NumberInput.svelte:101]: order

**Review by johanesa — APPROVED** [review]
lgtm

File: README.md (hunk 1 above)
Comment (Oli8) [reply README.md:103]: it's order alphabetically but there's a sub category for all the form components

File: src/lib/components/Form/NumberInput/NumberInput.svelte (hunk 2 above)
Comment (Oli8) [reply src/lib/components/Form/NumberInput/NumberInput.svelte:16]: `valid` is a prop function to add custom validation to the input
`isValid` is a boolean var whose value is computed by calling the `valid` function if provided

File: src/lib/components/Form/NumberInput/NumberInput.svelte (hunk 4 above)
Comment (Oli8) [reply src/lib/components/Form/NumberInput/NumberInput.svelte:101]: interesting point, but I havn't enforced such rule before and I don't feel like starting now :stuck_out_tongue:

**Pull request merged**